src/main/java/com/mparticle/kits/BranchUtil.java

```diff
@@ -0,0 +1,375 @@
+package com.mparticle.kits;
+
+import android.text.TextUtils;
+
+import com.mparticle.MPEvent;
+import com.mparticle.MParticle;
+import com.mparticle.commerce.CommerceEvent;
+import com.mparticle.commerce.Impression;
+import com.mparticle.commerce.Product;
+import com.mparticle.commerce.TransactionAttributes;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.branch.indexing.BranchUniversalObject;
+import io.branch.referral.Defines;
+import io.branch.referral.util.BRANCH_STANDARD_EVENT;
+import io.branch.referral.util.BranchEvent;
+import io.branch.referral.util.ContentMetadata;
+import io.branch.referral.util.CurrencyType;
+import io.branch.referral.util.ProductCategory;
+
+/**
+ * Created by sojanpr on 4/11/18.
+ * <p>
+ * Class for Branch utility methods to convert MParticle events to Branch events
+ * </p>
+ */
+
+class BranchUtil {
+    enum MPEventKeys {
+        position,
+        amount,
+        screen_name,
+        impression,
+        product_list_name,
+        product_list_Source,
+        checkout_options,
+        checkout_step,
+
+    }
+
+    enum ExtraBranchEventKeys {
+        product_category
+    }
+
+    private final HashMap<String, BRANCH_STANDARD_EVENT> BranchMParticleEventNames;
+
+    BranchUtil() {
+        BranchMParticleEventNames = new HashMap<>();
+        // Mapping MParticle Commerce Event names to corresponding Branch events
+        BranchMParticleEventNames.put("add_to_cart", BRANCH_STANDARD_EVENT.ADD_TO_CART);
+        BranchMParticleEventNames.put("add_to_wishlist", BRANCH_STANDARD_EVENT.ADD_TO_WISHLIST);
+        BranchMParticleEventNames.put("checkout", BRANCH_STANDARD_EVENT.INITIATE_PURCHASE);
+        BranchMParticleEventNames.put("click", BRANCH_STANDARD_EVENT.VIEW_ITEM);
+        BranchMParticleEventNames.put("purchase", BRANCH_STANDARD_EVENT.PURCHASE);
+        BranchMParticleEventNames.put("view_detail", BRANCH_STANDARD_EVENT.VIEW_ITEM);
+        BranchMParticleEventNames.put("checkout_option", BRANCH_STANDARD_EVENT.INITIATE_PURCHASE);
+
+        // Following MParticle events will be considered as Branch custom events since there is no matching events.
+        // "remove_from_cart", "remove_from_wishlist", "refund";
+
+        // Mapping MParticle Event names to corresponding Branch events
+        // Unknown,
+        // Navigation,
+        // Location,
+        BranchMParticleEventNames.put(MParticle.EventType.Search.name(), BRANCH_STANDARD_EVENT.SEARCH);
+        BranchMParticleEventNames.put(MParticle.EventType.Transaction.name(), BRANCH_STANDARD_EVENT.PURCHASE);
+        BranchMParticleEventNames.put(MParticle.EventType.UserContent.name(), BRANCH_STANDARD_EVENT.VIEW_ITEM);
+        // UserPreference
```

Comment:
since the string literal "add_to_cart", "add_to_wishlist" etc are available as constants in Product.ADD_TO_CART, Product.ADD_TO_WISHLIST, etc, I would rather you choose to either use all string literals or all constants (preferably constants) here